src/Platforms/AbstractPlatform.php

```diff
@@ -3117,6 +3117,14 @@ public function supportsColumnLengthIndexes(): bool
         return false;
     }
 
+    /**
+     * Whether the platform supports operator classes for indexes.
+     */
+    public function supportsIndexOperatorClasses(): bool
```

Comment:
I'm really hesitant about accepting an API that requires a "supports" method and is applicable to a single platform. I'd rather see the schema API being extensible so that the consumers can implement the features specific to their platforms w/o introducing core API changes.

For instance, currently, all the features of `Index` are implemented as "options" which are `array<string,mixed>`. This API is far from perfect because if there's a typo in the option name, it will be silently ignored.

There are some ideas proposed to improve this in https://github.com/doctrine/dbal/issues/3731 and https://github.com/doctrine/dbal/issues/4772. Although those are rather long-term projects and they won't necessarily cover this case, they could be used as inspiration for a better API for this feature.